protocol/daemons/bridge/client/sub_task_runner.go

```diff
@@ -0,0 +1,148 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"github.com/cometbft/cometbft/libs/log"
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	"github.com/dydxprotocol/v4-chain/protocol/daemons/bridge/api"
+	"github.com/dydxprotocol/v4-chain/protocol/daemons/bridge/client/types"
+	"github.com/dydxprotocol/v4-chain/protocol/daemons/constants"
+	libeth "github.com/dydxprotocol/v4-chain/protocol/lib/eth"
+	"github.com/dydxprotocol/v4-chain/protocol/lib/metrics"
+	bridgetypes "github.com/dydxprotocol/v4-chain/protocol/x/bridge/types"
+	eth "github.com/ethereum/go-ethereum"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	ethrpc "github.com/ethereum/go-ethereum/rpc"
+	"math/big"
+	"time"
+)
+
+type SubTaskRunner interface {
+	RunBridgeDaemonTaskLoop(
+		ctx context.Context,
+		logger log.Logger,
+		ethClient types.EthClient,
+		queryClient bridgetypes.QueryClient,
+		serviceClient api.BridgeServiceClient,
+	) error
+}
+
+type SubTaskRunnerImpl struct{}
+
+var _ SubTaskRunner = (*SubTaskRunnerImpl)(nil)
+
+// RunBridgeDaemonTaskLoop does the following:
+// 1) Fetches configuration information by querying the gRPC server.
+// 2) Fetches Ethereum events from a configured Ethereum client.
+// 3) Sends newly-recognized bridge events to the gRPC server.
+func (s *SubTaskRunnerImpl) RunBridgeDaemonTaskLoop(
```

Comment:
Refactored the RunBridgeDaemonTaskLoop method to be a member of the SubTaskRunnerImpl struct so that I can mock this implementation for testing.